### DISCUSSION TITLE
Adding multiple wells without a Menyanthes

### DISCUSSION THREAD

**Discussion by KatPathat** [pastas/pastas#276] [Q&A] [answered]
Hello,

I should start by stating that I am extremely new at using python or coding of any kind. 
My problem is that I have multiple extraction points (12 in fact) and I want to model using PASTAS, in one example a Menyanthes file is used which is not something I have. Would I have to therefore have to create multiple response functions to model all of the extraction wells? And how do I take the distance into account?

I appreciate any help I can get :)

/a super lost student

**Comment by dbrakenhoff** [comment, marked as answer]
Hi @KatPathat,

Always a good idea to ask for help! We're glad to help where we can.

The example you're referring to reads the data from a Menyanthes file. There's nothing special about the file, it just contains the data for this specific example problem. In the example we extract the relevant timeseries from this file and then proceed to building Pastas models two different ways. First, we add each extraction as a separate StressModel with its own response function. Next, we use the WellModel to account for all the extractions using only one response function. In this case the distance between extraction and observation point is used to scale the response for each individual extraction.

So you can follow the example, but you'll have to load your own data. Two possible approaches that you could take here are:

1. Add each extraction as a separate stress to the Pastas timeseries model. This would give you 12 stresses, each with their own response function. 
2. Add the extractions using `WellModel`, which also takes the distances between extractions and the observation point into account. For this specific StressModel you have to use the `HantushWellModel` response function. 

In the second approach, we make the assumption that all wells have a similar response (i.e. they're all pumping in the same aquifer). Depending on the geology and the location and configuration of the wells (depth of well screen, etc.) this assumption might or might not be appropriate. In your case I would try both approaches and see how the resulting models differ, and then decide which one is a more realistic approximation of actual situation.

In sort of psuedo-code, approach 1 would maybe look something like this:
```python
# read data
obs  # observation timeseries
extractions # list of pandas.Series containing discharge per extraction well

ml = ps.Model(obs)  # create model
# loop and add each extraction separately
for i, ex in enumerate(extractions):
    sm = ps.StressModel(ex, rfunc=ps.Gamma, name=f"ex-{i}", settings="well", up=False)
    ml.add_stressmodel(sm)
ml.solve()
```

Approach 2:
```python
# read data 
obs  # observation timeseries
extractions # list of pandas.Series containing discharge per extraction well
distances # calculate distances between extractions wells and observation well, in the same order as extractions

ml = ps.Model(obs)  # create model
# add extractions in WellModel
wm = ps.WellModel(extractions, rfunc=ps.HantushWellModel, name="extractions", 
                  distances=distances, settings="well")
ml.add_stressmodel(wm)
ml.solve()
```

Hope this helps a bit to get you started?

**Reply by KatPathat** [reply]
Thank you so much! Yes it does get me started on the right path. 
I am wondering, however, how the second approach takes into account the distance in relation to the observation well? Do I just include the distance (not coordinates) in the cvs file?
Additionally, I am assuming that if I use the first approach I should input a calculated drawdown as a stress for each respective pump instead as this approach does not take into account distance and therefore I cannot input a monthly pumping rate. 

Thank you so much for all your help!
Sincerely 
Kat

**Reply by dbrakenhoff** [reply]
Yes, the second approach takes the distance into account. The distances are passed as a list of numbers. Take a look at the notebook [10_multiple_wells.ipynb](https://github.com/pastas/pastas/blob/master/examples/notebooks/10_multiple_wells.ipynb) to see how it works. Run it yourself and take a look at the distances variable.

In the first approach the model is free to determine the influence of each well individually. All you have to do is provide the pumping rate, and it will attempt to calculate the parameters in such a way that the simulated heads match the observed heads. This can result in physically infeasible results, which you will have to analyse yourself. A conclusion could be that using the distance is necessary to avoid getting unrealistic results. Or you might have to drop some wells that seemingly do not contribute to changes in the observed heads. Or it might work really well... 

Hope that helps! Happy modelling!

**Reply by KatPathat** [reply]
Thank you so much for your help! It worked well :)
